Backpack/src/main/java/net/skyscanner/backpack/contentbubble/BpkContentBubble.kt

```diff
@@ -0,0 +1,224 @@
+  }
+
+  private var _fitContent = false
+  /**
+   * When fitContent is true the view will ensure extra space is reserved for the bubble pointer to
+   * ensure all content is visible.
+   */
+  var fitContent: Boolean
+    get() = _fitContent
+    set(value) {
+      _fitContent = value
+      requestLayout()
+    }
+
+  private var _round = false
+  /**
+   * Specify if corner radius should be added or not.
+   */
+  var round: Boolean
+    get() = _round
+    set(value) {
+      _round = value
+      requestLayout()
+    }
+
+  private var _pointerPosition = PointerPosition.MIDDLE
+  /**
+   * Specify where the pointer should be rendered.
+   *
+   * @see [PointerPosition]
+   */
+  var pointerPosition: PointerPosition
+    get() = _pointerPosition
+    set(value) {
+      _pointerPosition = value
+      requestLayout()
+    }
+
+  init {
+    this.initialize(attrs, defStyleAttr)
+  }
+
+  private fun initialize(attrs: AttributeSet?, defStyleAttr: Int) {
+    context.theme.obtainStyledAttributes(attrs, R.styleable.BpkContentBubble, defStyleAttr, 0)
+      ?.use {
+        _fitContent = it.getBoolean(R.styleable.BpkContentBubble_contentBubbleFitContent, fitContent)
+        _round = it.getBoolean(R.styleable.BpkContentBubble_contentBubbleRound, round)
+
+        _pointerPosition = it.getInt(R.styleable.BpkContentBubble_contentBubblePointerPosition, pointerPosition.id)
+          .let(::mapXmlToPointerPosition) ?: pointerPosition
+      }
+
+    background = null
+    setWillNotDraw(false)
+    setPadding(0, 0, 0, 0)
+    // PorterDuffXfermode doesn't work in some devices when hardware acceleration is on
+    setLayerType(LAYER_TYPE_SOFTWARE, null)
+
+    pointerDrawable.apply {
+      setBounds(0, 0, intrinsicWidth, intrinsicHeight)
+    }
+
+    cornerRadiusDrawable.apply {
+      setBounds(0, 0, intrinsicWidth, intrinsicHeight)
+    }
+
+    setUpMasks()
+  }
+
+  override fun draw(canvas: Canvas) {
+    val count = canvas.saveCount
+
+    val width = width.toFloat()
+    val height = height.toFloat()
+
+    val pointerHalfWidth = pointerDrawable.intrinsicWidth / 2
+    val pointerOffset = if (layoutDirection == View.LAYOUT_DIRECTION_RTL) {
+      1 - pointerPosition.offset
+    } else {
+      pointerPosition.offset
+    }
+
+    val yStart = height - pointerDrawable.intrinsicHeight
+    val pointerXStart = width * pointerOffset - pointerHalfWidth
+    val pointerXEnd = width * pointerOffset + pointerHalfWidth
+
+    clipRect.set(0f, yStart, pointerXStart, height)
+    canvas.clipOutRectCompat(clipRect)
+
+    clipRect.set(width, yStart, pointerXEnd, height)
+    canvas.clipOutRectCompat(clipRect)
+
+    super.draw(canvas)
+    canvas.restoreToCount(count)
+
+    paint.xfermode = porterDuffXfermode
+    canvas.drawBitmap(mask, pointerXStart, yStart, paint)
+
+    if (round) {
+      drawRadiusMask(canvas)
+    }
+
+    paint.xfermode = null
+  }
+
+  override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
```

Comment:
If we use padding, the onMeasure override is not necessary.

Reply:
done